### PR TITLE
Describe processes in points

### PR DESCRIPTION
**Pull request by IwonaLanger**
**Description**

Changes proposed in this pull request:

- Arrange long descriptions of processes (provisioning, deprovisioning) in points describing steps
- Introduce the steps in the diagrams
- Language fixes

**Related issue(s)**
#362

### CODE REVIEW

File: docs/contributor/02-10-operations.md [hunk 1]
```diff
@@ -26,77 +26,77 @@ the prerequisites yourself.
...
+
+2. The BtpOperator reconciler picks up the created CR and determines whether it should be responsible for representing the
+module status. 
+3. The BtpOperator CR reflects the status of the operand, that is, SAP BTP Service Operator, only when it is
+the oldest CR present in the cluster. Otherwise, it is given the `Error` state (3a) with the condition reason `OlderCRExists` and the message containing details about the CR responsible for reconciling the operand.
+4. For the only or the oldest CR present in the cluster,  a finalizer is added, the CR is set to the `Processing` state, and the
 reconciliation proceeds.
-Otherwise, it is given the `Error` state with the condition reason `OlderCRExists` and message containing details
-about the CR responsible for reconciling the operand.
+5. The reconciler looks for a `sap-btp-manager` Secret in the `kyma-system` Namespace with the label `app.kubernetes.io/managed-by: kcp-kyma-environment-broker`. This Secret contains the Service
```
Comment (mmitoraj) [inline docs/contributor/02-10-operations.md:47]: Please join all parts of the sentence in one line (47+49+50..) to fix the formatting of the ordered list.

File: docs/contributor/02-10-operations.md [hunk 2]
```diff
@@ -26,77 +26,77 @@ the prerequisites yourself.
...
 
-Next, the reconciler looks for a `sap-btp-manager` Secret in the `kyma-system` Namespace with the label `app.kubernetes.io/managed-by: kcp-kyma-environment-broker`. This Secret contains the Service
 Manager credentials for SAP BTP Service Operator and should be delivered to the cluster by KEB. If the Secret is
-missing, an error is thrown, the reconciler sets `Warning` state (with the condition reason `MissingSecret`) in the CR and stops the reconciliation until the Secret
-is created. When the Secret is present in the cluster, the reconciler verifies whether it contains required data. The
+missing, an error is thrown (5a), and the reconciler sets the `Warning` state (with the condition reason `MissingSecret`) in the CR and stops the reconciliation until the Secret
+is created. 
+6. When the Secret is present in the cluster, the reconciler verifies whether it contains the required data. The
```
Comment (mmitoraj) [inline docs/contributor/02-10-operations.md:52]: Starting from point 6 the ordered list formatting is broken. Fixes to point 5 should make it work.

File: docs/contributor/02-10-operations.md [hunk 3]
```diff
@@ -26,77 +26,77 @@ the prerequisites yourself.
...
+   The command triggers the deletion of the module resources in the cluster. By default, the existing ServiceInstances or ServiceBindings block the deletion. To unblock it, you must remove the existing ServiceInstances and ServiceBindings. Then, after the reconciliation, the BTP Operator resource is gone.
 
-At first, the deprovisioning process tries to perform the deletion in a hard delete mode. It tries to delete all 
-Service Bindings and Service Instances across all Namespaces. The time limit for the hard delete is 20 minutes. 
-After this time, or in case of an error, the process goes into soft delete mode, which runs deletion of finalizers from Service Instances and Service Bindings.
+   You can force the deletion by adding this label to the BTP Operator resource:
+   ```
+   force-delete: "true"
+   ```
+   If you use the label, all the existing Service Instances and Service Bindings are deleted automatically.
```
Comment (mmitoraj) [inline docs/contributor/02-10-operations.md:88]:
```suggestion
   If you use the label, all the existing ServiceInstances and ServiceBindings are deleted automatically.
```

File: docs/contributor/02-10-operations.md [hunk 4]
```diff
@@ -26,77 +26,74 @@ the prerequisites yourself.
...
+key values should be empty. If some required data is missing, the reconciler throws an error (6a) with the message about
+missing keys/values, sets the CR in the `Error` state (reason `InvalidSecret`), and stops the reconciliation until there is a change in the required
 Secret.
-
-After checking the Secret, the reconciler proceeds to apply and delete operations of the [module resources](/module-resources).
-The `module-resources` directory is created by one of GitHub Actions and contains manifests for applying and deleting operations. See [workflows](04-10-workflows.md#auto-update-chart-and-resources) for more details.
-First, the reconciler deletes outdated module resources stored as manifests in [to-delete.yml](/module-resources/delete/to-delete.yml).
-When all outdated resources are deleted successfully, the reconciler prepares current resources from manifests in the [apply](/module-resources/apply) directory to be applied to the cluster.
+7. After checking the Secret, the reconciler applies and deletes operations of the [module resources](/module-resources).
```
Comment (szwedm) [inline docs/contributor/02-10-operations.md:54]: `the reconciler applies and deletes operations of the module resources` doesn't sound right. It performs apply and delete operations on the module resources.

File: docs/contributor/02-10-operations.md [hunk 5]
```diff
@@ -26,77 +26,74 @@ the prerequisites yourself.
...
-If the process succeeds, the finalizer on BtpOperator CR itself is removed and the resource is deleted.
-If an error occurs during the deprovisioning, state of BtpOperator CR is set to `Error`. 
+2. At first, the deprovisioning process tries to perform the deletion in a hard delete mode. It tries to delete all ServiceBindings and ServiceInstances across all Namespaces. The time limit for the hard delete is 20 minutes. 
+3. Then it checks if there are any leftover ServiceBindings or ServiceInstances. 
+4. The hard delete is unsuccessful if a timeout is reached, if some resources are still present, or in case of an error. Then the process goes into the soft delete mode.
+5. The soft delete mode begins with deleting SAP BTP Service Operator module deployment and webhooks.
+6. The reconciler removes finalizers from ServiceBindings and deletes the related Secrets.
+7. The reconciler checks if there are any ServiceBindings left.
+8. Then it removes finalizers from ServiceInstances.
+9. The last step in the hard delete mode is checking for any leftover ServiceInstances.
```
Comment (szwedm) [inline docs/contributor/02-10-operations.md:94]: soft delete mode

**Review by szwedm — APPROVED** [review]
/lgtm

**Review by mmitoraj — APPROVED** [review]

**Pull request merged**